gitclone/gitclone.go

```diff
@@ -0,0 +1,226 @@
+package gitclone
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/bitrise-io/envman/envman"
+	"github.com/bitrise-io/go-utils/command"
+	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-io/go-utils/log"
+)
+
+// Config is the git clone step configuration
+type Config struct {
+	RepositoryURL string `env:"repository_url,required"`
+	CloneIntoDir  string `env:"clone_into_dir,required"`
+	Commit        string `env:"commit"`
+	Tag           string `env:"tag"`
+	Branch        string `env:"branch"`
+
+	BranchDest      string `env:"branch_dest"`
+	PRID            int    `env:"pull_request_id"`
+	PRRepositoryURL string `env:"pull_request_repository_url"`
+	PRMergeBranch   string `env:"pull_request_merge_branch"`
+	ResetRepository bool   `env:"reset_repository,opt[Yes,No]"`
+	CloneDepth      int    `env:"clone_depth"`
+
+	BuildURL         string `env:"build_url"`
+	BuildAPIToken    string `env:"build_api_token"`
+	UpdateSubmodules bool   `env:"update_submodules,opt[yes,no]"`
+	ManualMerge      bool   `env:"manual_merge,opt[yes,no]"`
+}
+
+const (
+	trimEnding = "..."
+)
+
+func printLogAndExportEnv(gitCmd git.Git, format, env string, maxEnvLength int) error {
+	l, err := output(gitCmd.Log(format))
+	if err != nil {
+		return err
+	}
+
+	if (env == "GIT_CLONE_COMMIT_MESSAGE_SUBJECT" || env == "GIT_CLONE_COMMIT_MESSAGE_BODY") && len(l) > maxEnvLength {
+		tv := l[:maxEnvLength-len(trimEnding)] + trimEnding
+		log.Printf("Value %s  is bigger than maximum env variable size, trimming", env)
+		l = tv
+	}
+
+	log.Printf("=> %s\n   value: %s\n", env, l)
+	if err := exportEnvironmentWithEnvman(env, l); err != nil {
+		return fmt.Errorf("envman export, error: %v", err)
+	}
+	return nil
+}
+
+func exportEnvironmentWithEnvman(keyStr, valueStr string) error {
```

Comment:
use our package instead of reimplementing this: https://github.com/bitrise-io/go-steputils/blob/master/tools/tools.go#L10

Reply:
Done.